pkg/types/types.go

```diff
@@ -201,54 +200,39 @@ const (
 type EventCode string
 
 const (
-	CandidatesFoundCode    EventCode = "candidates-found"
-	CandidatesFilteredCode EventCode = "candidates-filtered"
-	StartedCode            EventCode = "started"
-	ConnectedCode          EventCode = "connected"
-	QueryAskedCode         EventCode = "query-asked"
-	QueryAskedFilteredCode EventCode = "query-asked-filtered"
-	ProposedCode           EventCode = "proposed"
-	AcceptedCode           EventCode = "accepted"
-	FirstByteCode          EventCode = "first-byte-received"
-	FailedCode             EventCode = "failure"
-	SuccessCode            EventCode = "success"
-	FinishedCode           EventCode = "finished"
+	CandidatesFoundCode          EventCode = "candidates-found"
+	CandidatesFilteredCode       EventCode = "candidates-filtered"
+	StartedCode                  EventCode = "started"
+	StartedFetchCode             EventCode = "started-fetch"
+	StartedFindingCandidatesCode EventCode = "started-finding-candidates"
+	StartedRetrievalCode         EventCode = "started-retrieval"
+	ConnectedToSPCode            EventCode = "connected-to-sp"
+	QueryAskedCode               EventCode = "query-asked"
+	QueryAskedFilteredCode       EventCode = "query-asked-filtered"
+	ProposedCode                 EventCode = "proposed"
+	AcceptedCode                 EventCode = "accepted"
+	FirstByteCode                EventCode = "first-byte-received"
+	FailedCode                   EventCode = "failed"
+	FailedRetrievalCode          EventCode = "failed-retrieval"
+	SuccessCode                  EventCode = "success"
+	FinishedCode                 EventCode = "finished"
 )
 
 type RetrievalEvent interface {
 	fmt.Stringer
+
 	// Time returns the time that the event occurred
 	Time() time.Time
 	// RetrievalId returns the unique ID for this retrieval
 	RetrievalId() RetrievalID
 	// Code returns the type of event this is
 	Code() EventCode
-	// Phase returns what phase of a retrieval this even occurred on
-	Phase() Phase
-	// PhaseStartTime returns the time that the phase started for this storage provider
-	PhaseStartTime() time.Time
 	// PayloadCid returns the CID being requested
```

Comment:
I wonder whether we even want this -- does every event actually have a real payload CID? If not, let's delete it.

Reply:
PayloadCid() is called during the following events
- StartedFindingCandidatesEvent during fetch progress output
- StartedFetchEvent in aggregate events to instantiate an aggregated event
- FailedRetrievalEvent when a retrieval fails
- General logging for all events everytime the subscriber event callback is called

It's also asserted on the generic RetrievalEvent during some of the tests, but I think that's more of a result of the function originally being a part of the RetrievalEvent type than it being required in those use cases.

My opinion is that it's fairly helpful to have for logging, as it's one of the identifying pieces of information that can be used to correlate logs.

Reply:
I attempted to extract payloadCid out of the base event, but got caught on some assignable candidate finder tests that use CID keyed maps to assert that specific candidates were found for a given CID. The entire test suite set would have needed to be rewritten. I've made those changes to the `feat/remove-cid-from-events` branch if we wanted to move forward with this work. I'm going to leave the payloadCid in the base event for this PR.